File: Day12/q1.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

def sortPackages(head):
    count = [0, 0, 0]
    curr = head
    while curr:
        count[curr.data] += 1
        curr = curr.next
    curr = head
    for i in range(3):
        while count[i]:
            curr.data = i
            curr = curr.next
            count[i] -= 1
    return head

def createLinkedList(arr):
    head = Node(arr[0])
    curr = head
    for val in arr[1:]:
        curr.next = Node(val)
        curr = curr.next
    return head

def printList(head):
    while head:
        print(head.data, end=" → " if head.next else "")
        head = head.next
    print()


arr = list(map(int, input("Enter package colors (0/1/2) separated by space: ").split()))
head = createLinkedList(arr)


sorted_head = sortPackages(head)
print("Sorted packages:")
printList(sorted_head)

File: Day10/q1.java

import java.util.*;
class Node{
  int val;
  Node next;
  Node(int v){val=v;}
}
  public class q1{
    public static void main(String[] args){
      Scanner sc=new Scanner(System.in);
      int n =sc.nextInt();
      if(n==0) return ;
      Node head=new Node(sc.nextInt()),temp=head;
      for(int i=1;i<n;i++){
        temp.next=new Node(sc.nextInt());
        temp=temp.next;
      }
      Node curr=head;
      while(curr != null && curr.next != null){
        if(curr.val==curr.next.val)
          curr.next=curr.next.next;
        else
          curr=curr.next;
      }
      while (head !=null){
        System.out.print(head.val + (head.next !=null ? " -> ": ""));
        head=head.next;
      }
      
    }
  }










